lua/null-ls/builtins/diagnostics/golangci_lint.lua

```diff
@@ -20,39 +20,20 @@ return h.make_builtin({
         ignore_stderr = true,
         multiple_files = true,
         cwd = h.cache.by_bufnr(function(params)
-            -- find the nearest config and use that directory since v2 defaults to
-            -- "relative-path-mode: cfg" which reports file names relative to
-            -- the directory of the config file
-            local cfg_path_yml = u.root_pattern(".golangci.yml")(params.bufname)
-            if cfg_path_yml then
-                return cfg_path_yml
-            end
-            local cfg_path_yaml = u.root_pattern(".golangci.yaml")(params.bufname)
-            if cfg_path_yaml then
-                return cfg_path_yaml
-            end
-            local cfg_path_toml = u.root_pattern(".golangci.toml")(params.bufname)
-            if cfg_path_toml then
-                return cfg_path_toml
-            end
-            local cfg_path_json = u.root_pattern(".golangci.json")(params.bufname)
-            if cfg_path_json then
-                return cfg_path_json
-            end
-            -- nil defaults to git root
-            return nil
+            -- there might be cases when it's needed to setup cwd manually:
+            -- check the golangci-lint docs for relative-path-mode.
+            -- usually projects contain settings in root so this is sane default.
+            return u.root_pattern("go.mod")(params.bufname)
         end),
         args = h.cache.by_bufnr(function(params)
             -- params.command respects prefer_local and only_local options
             local version = vim.system({ params.command, "version" }, { text = true }):wait().stdout
             -- from observation the version can be either v2.x.x or 2.x.x
```

Comment:
Can you inline some documentation here supporting this claim

Reply:
@IlyasYOY

Reply:
I'm sorry, didn't notice your message.

I simply reverted some parts of the previous fix. The latter one fixed working with multiple versions but DIRNAME was extra. Lnter worked on project with single package at the root (like [gofmt command - cmd/gofmt - Go Packages](https://pkg.go.dev/cmd/gofmt)) but must of them contain multiple packages (like [segmentio/golines: A golang formatter that fixes long lines](https://github.com/segmentio/golines?tab=readme-ov-file)).